x/evm/types/types.go

```diff
@@ -959,6 +960,52 @@ func NewCommandBatchMetadata(blockHeight int64, chainID sdk.Int, keyID multisig.
 	}, nil
 }
 
+func (m CommandBatchMetadata) ValidateBasic() error {
+	if m.Status == BatchNonExistent {
+		return errors.New("batch does not exist")
+	}
+
+	if len(m.ID) != 32 {
```

Comment:
Not entirely convinced myself either, but I'm wondering if should we make sure `.ID` here isn't just filled with all zeros.

Reply:
yeah makes sense

Reply:
I'll address this in a separate PR